src/COVID19/model.py

```diff
@@ -369,6 +369,23 @@ def set_demographic_household_table(self, df_demo_house):
         
         covid19.set_demographic_house_table( self.c_params, int(n_total),int(n_households), ID_c, ages_c, house_no_c )
 
+    def set_occupation_network_table(self, df_occupation_networks, df_occupation_network_properties):
+        n_total = len(df_occupation_networks.index)
+        if n_total != self.get_param("n_total"):
+            raise ParameterException("df_occupation_networks must have n_total rows")
+
+        n_networks = df_occupation_networks['network_no'].max() + 1
+        covid19.set_occupation_network_table(self.c_params, int(n_total), int(n_networks))
+        [covid19.set_indiv_occupation_network_property(
+            self.c_params, int(row[0]),  int(row[1]), float(row[2]), float(row[3]), int(row[4]),
+            str(row[5]))
+            for row in df_occupation_network_properties[[
+            'network_no',  'age_type', 'mean_work_interaction', 'lockdown_multiplier', 'network_id',
+            'network_name']].values]
+
+        result = [covid19.set_indiv_occupation_network_table(self.c_params, int(row[0]), int(row[1])) for row in
```

Comment:
Calling this function through the Swig interface once per person is inefficient (if I change a test to 100k people it is very slow). It's quicker to declare arrays which can be read by both the C and Python and then call the function once. Look in the function above where we pass the demographic table in. This is done in one shot by declaring arrays with `ID_c =covid19.longArray(n_total)` which speeds it up.

Reply:
Done.